### PR TITLE
EVG-13365 support multiple sorts for patch tasks

### PR DESCRIPTION
**Pull request by john-m-liu**
- Add a new computed field in the task data structure that contains info about its base task. Right now it just includes ID and status but it's easy to add to
- Use this new field in the patchTasks query instead of the old method of adding the base status afterwards. This allows us to be able to filter and sort by base task data.
- Add a new sort parameter to the patchTask query that supports sorting by any of the columns. The old sorting params are kept for 1 deploy to maintain backwards compatibility, but will be removed ASAP (along with supporting code/tests)

### CODE REVIEW

File: graphql/resolvers.go [hunk 1]
```diff
@@ -1103,12 +1099,35 @@ func (r *queryResolver) PatchTasks(ctx context.Context, patchID string, sortBy *
...
 	}
-	tasks, count, err := r.sc.FindTasksByVersion(patchID, sorter, statusesParam, variantParam, taskNameParam, sortDirParam, pageParam, limitParam, []string{})
+	var taskSorts []task.TasksSortOrder
+	if len(sorts) > 0 {
+		taskSorts = []task.TasksSortOrder{}
+		for _, singleSort := range sorts {
+			key := ""
+			switch singleSort.Key {
+			// the keys here should be the keys for the column headers of the tasks table
+			case "NAME":
```
Comment (khelif96) [inline graphql/resolvers.go:1109]: You should use one of these enums. https://github.com/evergreen-ci/evergreen/blob/54ee8f373872e48d07597e7c8da479bd3a813e48/graphql/models_gen.go#L526

They are defined in `schema.graphql` and types are generated for both the frontend and backend.

File: graphql/schema.graphql [hunk 2]
```diff
@@ -299,6 +300,11 @@ input IssueLinkInput {
   issueKey: String!
 }
 
+input SortOrder {
+  Key: String!
+  Direction: Int!
```
Comment (khelif96) [inline graphql/schema.graphql:305]: This should be `TaskSortCategory` and `SortDirection` respectively since they are already defined enums.

Comment (khelif96) [reply graphql/schema.graphql:305]: This will allow us to use the same generated constant types on the frontend.

**Review by khelif96 — CHANGES_REQUESTED** [review]

File: graphql/resolvers.go (hunk 1 above)
Comment (john-m-liu) [reply graphql/resolvers.go:1109]: fixed

File: graphql/schema.graphql (hunk 2 above)
Comment (john-m-liu) [reply graphql/schema.graphql:305]: fixed

**Review by khelif96 — APPROVED** [review]

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**